robot_controllers/src/diff_drive_base.cpp

```diff
@@ -112,11 +110,67 @@ int DiffDriveBaseController::init(ros::NodeHandle& nh, ControllerManager* manage
   nh.param<double>("timeout", t, 0.25);
   timeout_ = ros::Duration(t);
 
-  // Get limits of base controller
-  nh.param<double>("max_velocity_x", max_velocity_x_, 1.0);
-  nh.param<double>("max_velocity_r", max_velocity_r_, 4.5);
-  nh.param<double>("max_acceleration_x", max_acceleration_x_, 0.75);
-  nh.param<double>("max_acceleration_r", max_acceleration_r_, 3.0);
+  // Get diff drive parameters
+  DiffDriveLimiterParams params = DiffDriveLimiter::getDefaultParams();
+  nh.getParam("track_width", params.track_width);
+  nh.getParam("max_velocity_x", params.max_linear_velocity);
+  nh.getParam("max_velocity_r", params.max_angular_velocity);
+  nh.getParam("max_acceleration_x", params.max_linear_acceleration);
+  nh.getParam("max_acceleration_r", params.max_angular_acceleration);
+
+  bool angular_velocity_limits_linear_velocity = params.angular_velocity_limits_linear_velocity;
+  nh.getParam("angular_velocity_limits_linear_velocity", angular_velocity_limits_linear_velocity);
+  params.angular_velocity_limits_linear_velocity = angular_velocity_limits_linear_velocity;
+
+  bool scale_to_wheel_velocity_limits = params.scale_to_wheel_velocity_limits;
+  nh.getParam("scale_to_wheel_velocity_limits", scale_to_wheel_velocity_limits);
+  params.scale_to_wheel_velocity_limits = scale_to_wheel_velocity_limits;
```

Comment:
why is this and the one above so much more complex than the others? does bool not translate from message to C++ cleanly?